src/data/fetchFromRmp.ts

```diff
@@ -0,0 +1,91 @@
+import fetch from "node-fetch"
+import {HEADERS, PROFESSOR_QUERY} from "~data/fetch";
+
+function getProfessorUrl(professorName: string, schoolId: string): string {
+    return `https://www.ratemyprofessors.com/search/teachers?query=${encodeURIComponent(professorName)}&sid=${btoa(`School-${schoolId}`)}`
+}
+function getProfessorUrls(professorNames: string[], schoolId: string): string[] {
+    const professorUrls = []
+    for (let i = 0; i < professorNames.length; i++) {
+        professorUrls.push(getProfessorUrl(professorNames[i], schoolId))
+    }
+    return professorUrls
+}
+
+function getProfessorIds(texts: string[], professorNames: string[]): string[] {
+    const professorIds = []
+    texts.forEach(text => {
+        const regex = /"legacyId":(\d+).*?"firstName":"(\w+)","lastName":"(\w+)"/g;
+        for (const match of text.matchAll(regex)) {
+            if (professorNames.includes(match[2] + " " + match[3])) {
+                professorIds.push(match[1]);
+            }
+        }
+    })
+    return professorIds
+}
+
+function getGraphQlUrlProp(professorId: string) {
+    HEADERS["Referer"] = `https://www.ratemyprofessors.com/ShowRatings.jsp?tid=${professorId}`
+    PROFESSOR_QUERY["variables"]["id"] = btoa(`Teacher-${professorId}`)
+    return {
+        method: "POST",
+        headers: HEADERS,
+        body: JSON.stringify(PROFESSOR_QUERY)
+    }
+}
+function getGraphQlUrlProps(professorIds: string[]) {
+    const graphQlUrlProps = []
+    professorIds.forEach(professorId => {
+        graphQlUrlProps.push(getGraphQlUrlProp(professorId))
+    })
+    return graphQlUrlProps
+}
+
+function fetchWithGraphQl(graphQlUrlProps: any[], resolve) {
+    const graphqlUrl = "https://www.ratemyprofessors.com/graphql";
+
```

Comment:
Try to avoid promise chaining, if at all possible. async await is cleaner, more maintainable, and has long term support and should be applicable here.